product/dialogs/miq_dialogs/miq_provision_google_dialogs_template.yaml

```diff
@@ -205,14 +205,18 @@
           :notes_display: :show
         :vm_name:
           :description: Instance Name
-          :required_method: :validate_vm_name
+          :required_method:
+            - :validate_vm_name
+            - :validate_regex
+          :required_regex: !ruby/regexp /\A(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)\Z/
+          :required_regex_fail_details: The name must be composed only of lower-case English alphabet characters (a-z), numbers (0-9), and dashes (-). The first character must be a lower-case English alphabet character, and the last character must not be a dash.
```

Comment:
I guess the obvious downside is that the regex then allows illegal characters that are only needed for automate; perhaps we are ok with this because its very unlikely that the customer will write $n{} in their hostname, but I'd prefer to avoid complicating the regex if needed.